de.metas.contracts/src/main/java/de/metas/contracts/impl/FlatrateBL.java

```diff
@@ -122,6 +124,8 @@
 	 */
 	public static final String MSG_HasOverlapping_Term = "de.metas.flatrate.process.C_Flatrate_Term_Create.OverlappingTerm";
 
+	public static final String MSG_INFINITE_LOOP = "de.metas.contracts.impl.FlatrateBL.extendContract.seenFlatrateConditionIds";
```

Comment:
pls give a better name for that AD_Message.
`seenFlatrateConditionIds` gives u no clue.
Try something like `InfinitLoopError' or something...